### DISCUSSION TITLE
Bibliography missing

### DISCUSSION THREAD

**Discussion by ikeronnyk** [joaomlourenco/novathesis#293] [Q&A] [answered]
Dear All, Good Evening!

Please I cannot see my list of Bibliography. May I please know how I can get this done in order to have my Bibliography.

Thank you.

**Comment by joaomlourenco** [comment]
Does your document have citations?  Bibliography only show up if you cite the documents.

**Reply by ikeronnyk** [reply]
Yes, Prof, my document has citations
<img width="1440" alt="citation" src="https://user-images.githubusercontent.com/24888452/152066413-8b80e097-542d-44ae-8596-a8621371a910.png">
.

**Comment by joaomlourenco** [comment, marked as answer]
Then follow the instructions in this link…
https://github.com/joaomlourenco/novathesis/wiki#warning

**Reply by ikeronnyk** [reply]
Thank you, Prof. I have had it changed but yet still the Bibliography does not appear
![biber](https://user-images.githubusercontent.com/24888452/152069474-a7ccd06e-fc18-40d1-890e-2ecc0cd45b3a.png)
.

**Reply by ikeronnyk** [reply]
Sorry, this instead.

Thank you
<img width="1440" alt="Biber main" src="https://user-images.githubusercontent.com/24888452/152069721-a5ac1f37-26d6-4a2f-91b1-b277807610bf.png">

**Reply by ikeronnyk** [reply]
Now changed to Latexmk.


Thank you.
![latexmk](https://user-images.githubusercontent.com/24888452/152070593-37d0ac40-ff6e-43fd-859e-a4b661e227d9.png)

**Reply by ikeronnyk** [reply]
Sorry, Prof. I think this is the right screenshot for the Latexmk.

Thank you
<img width="1440" alt="Untitled 5" src="https://user-images.githubusercontent.com/24888452/152070952-6bdfb2af-2b7a-4eb5-ba7a-ca340accc325.png">
.

**Reply by ikeronnyk** [reply]
Dear Prof,

I did as you requested and had the underlisted information but do not know what that means because I have reference information in the Bibliography.


Package biblatex Info: Trying to load bibliographic data...
Package biblatex Info: ... file 'template.bbl' found.
(./template.bbl)

Package biblatex Warning: citation 'bishop2007discrete' was not found at all in
 the .bbl, bibliography labels cannot be generated, please correct the citation
.


Package biblatex Warning: citation 'agresti2013categorical' was not found at al
l in the .bbl, bibliography labels cannot be generated, please correct the cita
tion.


Package biblatex Warning: citation 'agresti2010analysis' was not found at all i
n the .bbl, bibliography labels cannot be generated, please correct the citatio
n.


Package biblatex Warning: citation '1_tang2012applied' was not found at all in 
the .bbl, bibliography labels cannot be generated, please correct the citation.



Package biblatex Warning: citation 'fagerland2017statistical' was not found at 
all in the .bbl, bibliography labels cannot be generated, please correct the ci
tation.


Package biblatex Warning: citation 'lloyd1999statistical' was not found at all 
in the .bbl, bibliography labels cannot be generated, please correct the citati
on.


Package biblatex Warning: citation 'kateri2014contingency' was not found at all
 in the .bbl, bibliography labels cannot be generated, please correct the citat
ion.


Package biblatex Warning: citation 'kateri2008statistical' was not found at all
 in the .bbl, bibliography labels cannot be generated, please correct the citat
ion.


Package biblatex Warning: citation 'birch1963maximum' was not found at all in t
he .bbl, bibliography labels cannot be generated, please correct the citation.

**Comment by joaomlourenco** [comment]
Then open `template.log` in you favorite text editor and search for these:
```
Error
error
Warning
warning
```
There must be something related to bibliography.

**Comment by ikeronnyk** [comment]
Dear Prof,
Thank you for your help yesterday. After several manipulations, I have been able to figure it out and I now have the Bibliography. The only challenge is, it numbers some of the references as [0]. Is there a way I can overcome this?

Kindest regards.

**Reply by TyredMayfly** [reply]
Could you elaborate on your process? I seem to have the same problem, but can't get it to work...

**Reply by joaomlourenco** [reply]
When all the references are [0], it means that either:

1. There is an error processing the bibliography and LaTeX is not finishing the process; or
2. LaTeX is “confused” with the auxiliary files.

To eliminate 1), but be sure you have no errors in the LaTeX compiling and Biber execution (ared box with number in Overleaf)  .
To eliminate 2) from the equation, just delete all the temporary files (recomplie from scratch in Overleaf).

**Reply by TyredMayfly** [reply]
Hi Joao,

Thank you for the quick reply. I have about 8 chapters, and when I just import one, it functions fine. When I import three, the bibliography shows all numbers as [0] and when I import five or more the bibliographies dissapear (I show per chapter).  It shows warnings for all my text citations that there are no references found. There are no red boxes and I tried "Recompile from scratch". 

![image](https://github.com/joaomlourenco/novathesis/assets/56123998/7c05b574-3b54-4277-8f23-5707799ab1c6)

Thank you in advance!

**Reply by joaomlourenco** [reply]
Definitely (99,99% confidence) the problem is somewhere in your text.
Try to import just the "third" chapter where the problem shows up.  Does it work?

**Reply by TyredMayfly** [reply]
Thanks! I ended up linking it to a portion of the text, and I think there were one or two bibliography entries with errors in them that led to the problem. But I'm not a 100% sure, it works now, so hopefully it keeps on working. Thanks again!

**Comment by ikeronnyk** [comment]
Dear Prof,

Good Evening. 

Please how do I get all authors listed in a Bibliography and not et. al. when there are more than 3 authors. Thank you.
<img width="1440" alt="et  al" src="https://user-images.githubusercontent.com/24888452/173683544-56bfa6ad-53a9-4682-a11b-d4b3b98d15ac.png">

**Reply by ikeronnyk** [reply]
Like we have in number 5, I want the names of all the authors to appear.

Thank you.

**Reply by joaomlourenco** [reply]
Which bibliography style are you using?

**Reply by ikeronnyk** [reply]
I cannot tell but I have not changed the default style

**Reply by ikeronnyk** [reply]
Prof, please here are the codes from the biblatex file.

Thank you.


%%  BIBLIOGRAPHY CUSTOMIZATION / OPTIONS
%%============================================================

%%------------------------------------------------------------
% If it works, leave biber.  If it fails for you, try bibtex
%\ntbibsetup{backend=biber}
 \ntbibsetup{backend=bibtex}

%%------------------------------------------------------------
%% Print only initials of given name
\ntbibsetup{giveninits=true}

%%------------------------------------------------------------
%% At the end of each entry in the bibliography lists in which 
%% page(s) this entry is cited. Comment if you do not want this.
\ntbibsetup{backref=true, backrefstyle=three} 

%%------------------------------------------------------------
%% Show the dates in ISO format (YYY-MM-DD)
\ntbibsetup{urldate=iso,date=iso}


%%============================================================
%% BIBLIOGRAPHY STYLE
%% For other styles, see Section 3.3 in the BibLaTeX manual
%%============================================================

%%------------------------------------------------------------
%% NUMBERED ORDER BY LAST NAME OF FIRST AUTHOR - this is the default
% \ntbibsetup{style=numeric-comp, sortcites=true, sorting=nyt}

%%------------------------------------------------------------
%% NUMBERED ORDER BY CITATION
% \ntbibsetup{style=numeric-comp, sortcites=true, sorting=none}

%%------------------------------------------------------------
%% APLHA CITATION ORDER BY LAST NAME OF FIRST AUTHOR
% \ntbibsetup{style=alphabetic, sortcites=true, sorting=nyt}

%%------------------------------------------------------------
%% AUTHOR YEAR SORTED BY NAME AND YEAR
% \ntbibsetup{style=authoryear-comp, sortcites=true, sorting=nyt}

%%------------------------------------------------------------
%% APA-LIKE CITATION
% \ntbibsetup{backend=biber, natbib=true, style=apa6}

**Reply by joaomlourenco** [reply]
Open file `Config/2_biblatex.tex` and add a line with
```latex
\ntbibsetup{smaxbibnames=99}
```

**Reply by ikeronnyk** [reply]
Dear Prof, after adding the line of code, this is the output
<img width="1440" alt="bibtex error" src="https://user-images.githubusercontent.com/24888452/178158324-d54aca65-e826-425d-b06f-c4da143ff017.png">

**Reply by NoronhaC** [reply]
Sorry do I need to change something in the Template in the OverLeaf as well?
Regards
Pedro

**Reply by NoronhaC** [reply]
![image](https://github.com/joaomlourenco/novathesis/assets/135113777/23745838-ae65-49c1-adac-80e778b690f9)
Sorry I add my bibliography and it does not recognize, I also correct the artilce to article. Any idea on how to solve this issue? 
Thank you

**Reply by NoronhaC** [reply]
ok resolvideo, sorry, cite e nao ref
